File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/exception/ExceptionHandler.java
package ru.tggc.telegrambotcore.exception;

import com.pengrad.telegrambot.model.Chat;
import com.pengrad.telegrambot.model.User;
import ru.tggc.telegrambotcore.dto.Response;

public interface ExceptionHandler {

    Response handleException(Exception e, Chat chat, User from);

    String buildMessageToAdmin(String s, Chat chat, User from);
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/exception/ExceptionHandler.kt
package ru.tggc.telegrambotcore.exception

import com.pengrad.telegrambot.model.Chat
import com.pengrad.telegrambot.model.User
import ru.tggc.telegrambotcore.dto.Response

interface ExceptionHandler {
    fun handleException(e: Exception, chat: Chat, from: User): Response

    fun buildMessageToAdmin(s: String, chat: Chat, from: User): String
}
